script.js

```diff
@@ -0,0 +1,99 @@
+var imArr = [];
+var counter = 0;
+var nIntervId;
+
+
+$(function() {
+  // attach the form submission to the search function
+  $("#search-form").on('submit', search);
+  });
+
+function search(event) {
+// Stop the form from changing the page.
+  event.preventDefault();
+  clearSearchResults();
+
+  var input = $("#query")
+  var userQuery =  input.val();
+
+
+  $.get('https://www.reddit.com/search.json', {
+    q: userQuery,
+    limit: 100
+    // nsfw:no
+  }).done(function(response) {
+    console.log(response);
+  
+
+//changing based on Nat's code
+//this filters through the children to check for the preview tag & find the url
+//then pushes those urls into an array
+response.data.children.filter(function(result) {
+if(result.data.preview && result.data.preview.images[0].source.url) {
+    imArr.push(result.data.preview.images[0].source.url);
+    console.log(imArr);
+        } 
+      });
+    changeSlide();
+  });
+
+
+  // Clear previous search results.
+  function clearSearchResults() {
+    $("#results").html("");
+    $('#slide-show').html(""); 
+    imaArr= [];
+  }
+
+  //hides button and input
+  $('#submit').click(function (){
```

Comment:
Make sure this is in the document ready function: $(function(){....
otherwise the listener may or may not be added properly. Any listeners that should be added on page load go in document ready function.